error.go

```diff
@@ -0,0 +1,40 @@
+package xjson
+
+import (
+	"errors"
+)
+
+// Error implements a JSON-serializable error
+// "Err" for less confusion over func names
+type Error struct {
+	Err error
+}
+
+func NewError(e error) Error { return Error{Err: e} }
+
+func (xe *Error) Error() string { return xe.Err.Error() }
+
+func (xe Error) String() string {
+	return xe.Err.Error()
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (xe Error) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + xe.Err.Error() + `"`), nil
+}
+
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (xe *Error) UnmarshalJSON(b []byte) error {
+	s, err := stripQuotes(string(b))
+	if err != nil {
+		return err
+	}
+	if len(s) == 0 {
+		return errors.New("Error message must not be empty")
```

Comment:
nit: in idiomatic Go, error strings should always start with a lower-case letter when possible